modules/gcs/variables.tf

```diff
@@ -132,6 +132,40 @@ variable "notification_config" {
   default = null
 }
 
+variable "objects_to_upload" {
+  description = "Objects to be uploaded to bucket."
+  type = map(object({
+    name                = string
+    metadata            = optional(map(string))
+    content             = optional(string)
+    source              = optional(string)
+    cache_control       = optional(string)
+    content_disposition = optional(string)
+    content_encoding    = optional(string)
+    content_language    = optional(string)
+    content_type        = optional(string)
+    event_based_hold    = optional(bool)
+    temporary_hold      = optional(bool)
+    detect_md5hash      = optional(string)
+    storage_class       = optional(string)
+    kms_key_name        = optional(string)
+    customer_encryption = optional(object({
+      encryption_algorithm = optional(string)
+      encryption_key       = string
+    }))
+  }))
+  validation {
+    condition = alltrue([
+      for k, v in var.objects_to_upload :
+      v.content != null
```

Comment:
This condition would allow both content and source to be specified. I think you want something like `(v.content != null) != (v.source != null)`. In any case I think you this is checked by the provider, you don't have to duplicate it here.

Reply:
removed validation